src/init.cpp

```diff
@@ -196,8 +196,9 @@ void Shutdown()
     StopRPC();
     StopHTTPServer();
 #ifdef ENABLE_WALLET
-    if (pwalletMain)
-        pwalletMain->Flush(false);
+    for (CWallet_ptr pwallet : vpwallets) {
```

Comment:
I like the approach by @jonasschnelli better. Let's keep most of the multiwallet logic in `wallet.cpp`.